lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1204,41 +1204,96 @@ def cmd_route_tabs(str, words)
   end
 
   def cmd_save_help
-    print_line "Usage: save"
+    print_line "Usage:"
     print_line
-    print_line "Save the active datastore contents to disk for automatic use across restarts of the console"
+    print_line "save:                       save the active module options to disk for automatic use across restarts of the console"
+    print_line "save -d / --delete:         delete the active module options from disk without refreshing current options"
+    print_line "save -r / --reload:         reload default options for the active module"
+    print_line "save -l / --load:           load the saved options for the active module"
+    print_line "save -D / --delete-all:     delete ALL datastore options from the config file"
     print_line
     print_line "The configuration is stored in #{Msf::Config.config_file}"
     print_line
   end
 
   #
-  # Saves the active datastore contents to disk for automatic use across
+  # Deletes or saves the active datastore contents to disk for automatic use across
   # restarts of the console.
   #
   def cmd_save(*args)
-    # Save the console config
-    driver.save_config
-
-    begin
-      FeatureManager.instance.save_config
-    rescue StandardException => e
-      elog(e)
+    case args[0]
+    when '-D', '--delete-all'
+      # Delete all saved info
+      method = 'delete_all_module_configs'
+    when '-d', '--delete'
+      # Delete saved info for current module only
+      method = 'delete_module_config'
+    when '-h', '--help'
+      method = 'help'
+    when '-r', '--reload'
+      method = 'reload'
+    when '-l', '--load'
+      method = 'load'
+    when nil
+      # Save config for current module
+      method = 'module_save'
+    else
+      print_line('Unknown save option. Type `save -h` for more information.')
+      return false
     end
 
-    # Save the framework's datastore
-    begin
-      framework.save_config
+    case method
```

Comment:
I feel like two case statements is more complexity than what's required. Could you move the logic into the above `case` statement and seperate the larger blocks of code into their own methods?